homeassistant/components/hydrawise/binary_sensor.py

```diff
@@ -111,3 +139,28 @@ def available(self) -> bool:
         if self.entity_description.always_available:
             return True
         return super().available
+
+
+class HydrawiseZoneBinarySensor(HydrawiseBinarySensor):
+    """A sensor implementation for a Hydrawise irrigation zone."""
+
+    zone: Zone
+
+    async def start_watering(self, duration: timedelta | None = None) -> None:
+        """Start watering in the irrigation zone."""
+        if duration is None:
+            duration = timedelta(minutes=0)
+        if duration > MAX_WATERING_TIME:
+            LOGGER.warning("Clamping watering time to %s", MAX_WATERING_TIME)
```

Comment:
As already mentioned this should be handled in the schema and also in `services.yaml` so the user is limited within the allowed range.

Reply:
Done.